app/assets/javascripts/authoring/events.js

```diff
@@ -257,6 +258,77 @@ function createEvent(point, duration) {
     updateStatus();
 };
 
+function newEventObj(snapPoint, duration){
+    duration = duration || 60;
+    var startTimeObj = getStartTime(snapPoint[0]);
+    var newEvent = {
+        "title":"New Event", "id":createEventId(),
+        "x": snapPoint[0]-4, "min_x": snapPoint[0], "y": snapPoint[1], //NOTE: -4 on x is for 1/15/15 render of events
+        "startTime": startTimeObj["startTimeinMinutes"], "duration":duration,
+        "members":[], timer:0, task_startBtn_time:-1, task_endBtn_time:-1,
+        "dri":"", "pc":"", "notes":"", "startHr": startTimeObj["startHr"], "status":"not_started",
+        "startMin": startTimeObj["startMin"], "gdrive":[], "completed_x":null, "inputs":"", "all_inputs":"", "outputs":"", events_after : "",
+        "docQs": [["Please explain all other design or execution decisions made, along with the reason they were made",""],
+        ["Please add anything else you want other team members, the project coordinator, or the client, to know. (optional)",""]],
+        "outputQs":{},"row": Math.floor((snapPoint[1]-5)/_foundry.timeline.rowHeight)};
+
+    return newEvent;
+}
+function duplicateEventObj(eventObject){
```

Comment:
This seems like the function in which you would replace the google drive folder information with the new folder created for the duplicate event (e.g., newObject["gdrive"] should contain a new array with the new folder information instead of the gdrive information from the original event that was copied). See my comments on your PR for additional information.